packages/driver/src/cy/commands/waiting.ts

```diff
@@ -270,8 +281,51 @@ export default (Commands, Cypress, cy, state) => {
     })
   }
 
+  Cypress.primaryOriginCommunicator.on('wait:for:xhr', ({ args: [str, options] }, originPolicy) => {
+    options.isCrossOriginSpecBridge = true
+    waitString(null, str, options).then((responses) => {
+      // remove props that aren't serializable (these aren't used by consumers so are okay to remove)
+      const omitFn = (response) => _.omit(response, ['subscriptions', 'setLogFlag'])
+      const preprocessedResponses = _.isArray(responses) ? responses.map(omitFn) : omitFn(responses)
+
+      Cypress.primaryOriginCommunicator.toSpecBridge(originPolicy, 'wait:for:xhr:end', { responses: preprocessedResponses })
+    }).catch((err) => {
+      options._log?.error(err)
+      Cypress.primaryOriginCommunicator.toSpecBridge(originPolicy, 'wait:for:xhr:end', { err: preprocessForSerialization(err) })
+    })
+  })
+
+  const delegateToPrimaryOrigin = ([_subject, str, options]) => {
+    return new Promise((resolve, reject) => {
```

Comment:
now that we have a use case for promise based communicator events, I wonder if we can also add a generic method on the communicator to wrap an one time event in a promise and define some type of generic CB function in an `async`/`await` context, like:

```js
const promisifyToPrimary(toPrimaryEvent, toPrimaryEventArgs, fromPrimaryEvent, onFromPrimaryEventResponse){
  return new Promise((resolve, reject) => {
     Cypress.specBridgeCommunicator.once(fromPrimaryEvent, async (args) => {
         try {
             const genericResults = await onFromPrimaryEventResponse(args)
             resolve(genericResults)
         } catch(e){
             reject(e)
         }
     })
  })
  
  Cypress.specBridgeCommunicator.toPrimary(ToPrimaryEvent, args)
}
```

then you can just make the `onFromPrimaryEventResponse` generic, like

```js
    if (err) {
          if (options.log) {
            Cypress.state('onBeforeLog', (log) => {
              // skip this 'wait' log since it was already added through the primary
              if (log.get('name') === 'wait') {
                // unbind this function so we don't impact any other logs
                cy.state('onBeforeLog', null)

                return false
              }

              return
            })
          }

          throw err
        }

       return responses
```

Reply:
Typically I would wait until there is a second use-case before creating a generic pattern. Do we know of another use-case? Thoughts on implementing now or waiting?

Reply:
I'm OK with waiting. But I wonder if we want to document someplace that if we see this pattern creep up, that we implement it in the communicator

Reply:
Discussed in standup today and we are going to wait before creating a generic pattern. I don't think there is really a good spot to document this but something that we will keep in mind for the future.